modules/internal/ChapelBase.chpl

```diff
@@ -1293,12 +1311,15 @@ module ChapelBase {
   }
 
   inline proc %=(ref lhs:int(?w), rhs:int(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0 then
+        halt("Attempt to compute a modulus by zero");
     __primitive("%=", lhs, rhs);
   }
   inline proc %=(ref lhs:uint(?w), rhs:uint(w)) {
-    __primitive("%=", lhs, rhs);
-  }
-  inline proc %=(ref lhs:real(?w), rhs:real(w)) {
```

Comment:
Oh sorry, I didn't [re?-]read the PR message

And gosh, that's weird... I could've sworn someone asserted that % was supported on reals the other week, and I was surprised by that, but then realized that C supports fmod() and figured we must map % to fmod.

Since your PR isn't related to % on reals, I'd leave that overload in for now and then I think we should add an overload for % on reals as a separate PR (which I'm willing to drive if you're not interested).